src/coreclr/src/tools/Common/Compiler/HardwareIntrinsicHelpers.cs

```diff
@@ -40,6 +40,7 @@ public static bool IsHardwareIntrinsic(MethodDesc method)
             return false;
         }
 
+#if !READYTORUN
```

Comment:
Why doesn't this logic apply to R2R? Isn't it supposed to be the CPUID caching/checking logic for R2R/Crossgen?

Reply:
Things outside READYTORUN are used in CoreRT. CoreRT doesn't block pregeneration of `Vector<T>` like crossgen2 does (it can't, because it's AOT) - AFAIK enabling SSE4.X would result in Vector using them. So they're known at compile time as never supported.

Maybe it would be better to delete this code and make `HardwareIntrinsicHelpers` a partial class instead so that it doesn't distract within the shipping product.

Reply:
That seems like something that should be handled in the `Compiler::getSIMDSupportLevel()` check, rather than something which blocks `if (Sse41.IsSupported) { }` from being compiled down to a cached CPUID check like SSE3 and the others have happen.

Reply:
Substantial engineering for CoreRT is not budgeted for at this time. The current logic works well enough, and while I agree in the long term this isn't ideal, it should work for now.